pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -57,6 +57,13 @@ public PgDatabaseMetaData(PgConnection conn) {
   private int nameDataLength; // length for name datatype
   private int indexMaxKeys; // maximum number of keys in an index.
 
+  private byte[] getCatalogName(@Nullable String catalog) throws SQLException {
+    if (catalog == null) {
+      return connection.getCatalog().getBytes();
```

Comment:
We should use connection's encoding here. Otherwise there's a risk the bytes will be decoded differently